Core/Src/bms.c

```diff
@@ -47,17 +47,16 @@ void bms_init()
 	assert(bms_monitor_queue);
 }
 
-void vBMSCANMonitor(void *pv_params)
+void vAMSCANMonitor(void *pv_params)
 {
 	can_msg_t msg_from_queue;
 
 	osTimerStart(bms->bms_monitor_timer, BMS_CAN_MONITOR_DELAY);
 	for (;;) {
-		if (osOK == osMessageQueueGet(bms_monitor_queue,
-					      &msg_from_queue, NULL,
-					      osWaitForever)) {
-			osTimerStart(bms->bms_monitor_timer,
-				     BMS_CAN_MONITOR_DELAY);
-		}
+		osThreadFlagsWait(NEW_AMS_MSG_FLAG, osFlagsWaitAny,
```

Comment:
why is this needed?  I thought queues took care of waiting for us as long as we do wait forever.